### ISSUE TITLE
IOS FxAccountManagerTests failure on Circle CI

### ISSUE THREAD

**Issue by tarikeshaq** [mozilla/application-services#3233]
Filing after #3232,
From the Circle CI failure:
```
 issue MozillaAppServicesTests.FxAccountManagerTests
 
  testAccountRestorationEnsureCapabilitiesNonAuthError, XCTAssertTrue failed
 
  /Users/distiller/project/megazords/ios/MozillaAppServicesTests/FxAccountManagerTests.swift:193
    XCTAssertTrue(account.invocations.contains(MockFxAccount.MethodInvocation.registerPersistCallback))
 
        XCTAssertTrue(account.invocations.contains(MockFxAccount.MethodInvocation.getProfile))
 
 ```

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/SYNC-1392)
┆Story Points: 3
┆Sprint: SYNC - end 2020-07-17

**Comment by tarikeshaq** [comment]
Had another failure recently that passed after re-run: https://app.circleci.com/pipelines/github/mozilla/application-services/12011/workflows/6cc29295-9002-4183-8518-172873219f46/jobs/72569/steps

Here are the logs:
```
MozillaAppServicesTests.FxAccountManagerTests
 
  testAccountRestorationEnsureCapabilitiesNonAuthError, XCTAssertTrue failed
 
  /Users/distiller/project/megazords/ios/MozillaAppServicesTests/FxAccountManagerTests.swift:193

 
        XCTAssertTrue(account.invocations.contains(MockFxAccount.MethodInvocation.registerPersistCallback))
 
        XCTAssertTrue(account.invocations.contains(MockFxAccount.MethodInvocation.getProfile))

 

 

 
	 Executed 20 tests, with 1 failure (0 unexpected) in 1.527 (1.587) seconds
 

 
2020-07-06 14:49:40.779 xcodebuild[1442:8348] [MT] IDETestOperationsObserverDebug: 172.910 elapsed -- Testing started completed.
 
2020-07-06 14:49:40.779 xcodebuild[1442:8348] [MT] IDETestOperationsObserverDebug: 0.000 sec, +0.000 sec -- start
 
2020-07-06 14:49:40.780 xcodebuild[1442:8348] [MT] IDETestOperationsObserverDebug: 172.910 sec, +172.909 sec -- end
 
Failing tests:
 
	MozillaAppServicesTests:
 
		FxAccountManagerTests.testAccountRestorationEnsureCapabilitiesNonAuthError()
 

 
** TEST FAILED **
 
```

**Comment by tarikeshaq** [comment]
Possible fix merged #3334 I'm keeping this open for the next short while to keep an eye on our CI tests. Will close this issue if the test failures are gone

**Comment by tarikeshaq** [comment]
Been a week, haven't seen CI fail on the same error. Closing this for now
